index.js

```diff
@@ -22,40 +33,53 @@ module.exports = function(source, sourceMap) {
         this.cacheable();
     }
 
-    if (Object.keys(query).length) {
-        var inject = '\n/* injects from baggage-loader */\n';
+    var filePaths = Object.keys(query);
+    var injections = [];
+    if (filePaths.length) {
+      injections = filePaths.map(function(filePath) {
+
+        var varName;
+        var loadersForFile = '';
+        var inject = null;
+
+        if (typeof query[filePath] === 'object') {
+          var fileConfig = query[filePath];
+          var loaderStringForFile = fileConfig.loaders || '';
+          loadersForFile = loaderStringForFile.replace(/\*/g, '!') + '!';
```

Comment:
if you don't conditionally apply this, the loader will prefix all imports with `!` voiding your webpack config's rules. (I've merged your branch into my fork and fixed it)